bot/exts/events/advent_of_code/_cog.py

```diff
@@ -133,6 +135,7 @@ async def about_aoc(self, ctx: commands.Context) -> None:
         """Respond with an explanation of all things Advent of Code."""
         await ctx.send(embed=self.cached_about_aoc)
 
+    @commands.guild_only()
     @adventofcode_group.command(name="join", aliases=("j",), brief="Learn how to join the leaderboard (via DM)")
```

Comment:
See the function docstring.
```py
"""DM the user the information for joining the Python Discord leaderboard."""
```
The "via DM" is in reference to the fact that it does `ctx.author.send(the_information)` rather than the fact the command gets invoked in DMs.

I made this `guild_only` since it uses `Member`-only attributes, such as the `.roles` check.